test/integration/file_artifact_test.js

```diff
@@ -228,4 +231,67 @@ suite('artifact extration tests', () => {
     // Missing artifact should have an error...
     assert.equal(result.artifacts['public/my-missing.txt'].storageType, 'error');
   });
+
+  test('upload retry', async () => {
+    let worker = new TestWorker(DockerWorker);
+    await worker.launch();
+
+    worker.on('task run', function() {
+      iptables.reject({
```

Comment:
this test doesn't work... you're not reading anything from the stream... So you don't test retries...
To make this work you need to read half the stream and then reset...

This will show you that actually can't retry uploading a stream, as reading from stream has side-effects... And our streams can't seek... So we probably need to:
 A) store in a temporary file
 B) reload from docker container (but only reload the file that failed to upload)

(B) could get complicated...